src/ProjectOrigin.VerifiableEventStore/Services/EventStore/MemoryEventStore.cs

```diff
@@ -11,6 +11,8 @@ public class MemoryEventStore : IEventStore
 {
     private readonly List<BatchWrapper> _batchWrappers = new();
     private readonly int _batchSize;
+    private readonly object _lockObject = new();
```

Comment:
Did you think about using a SemaphoreSlim over a lock?

Reply:
Kinda, since performance is  not the main focus of the memory implementation, where async SemaphoreSlim and async would insure that threads could be released to other work, i tought the more simple look of the code was better in this case for readability, since a semaphore would require a try-finally to ensure it being released.

```csharp
public Task Store(VerifiableEvent @event)
{
    lock (_lockObject)
    {
       ...
    }
}

public Task Store(VerifiableEvent @event)
{
    await semaphore.WaitAsync();

    try
    {
        ...
    }
    finally
    {
        semaphore.Release();
    }
}
```